src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/FileExistsValidator.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Threading.Tasks;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a recipe or deployment bundle option with a FilePath typehint points to an actual file.
+    /// This can either be an absolute path to the file or relative to the project path
+    /// </summary>
+    public class FileExistsValidator : IOptionSettingItemValidator
+    {
+        private readonly IFileManager _fileManager;
+
+        public FileExistsValidator(IFileManager fileManager)
+        {
+            _fileManager = fileManager;
+        }
+
+        public string ValidationFailedMessage { get; set; } = "The specified file does not exist";
+
+        public Task<ValidationResult> Validate(object input, Recommendation recommendation)
+        {
+            var inputFilePath = input?.ToString() ?? string.Empty;
```

Comment:
Similar to my other comment, there should be an AllowEmpty attribute which determines if the path can be empty or not.